interfaces/builtin/docker.go

```diff
@@ -0,0 +1,185 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+var dockerPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as the Docker daemon. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+# Allow sockets
+/{,var/}run/docker.sock rw,
+/{,var/}run/docker/     rw,
+/{,var/}run/docker/**   mrwklix,
+/{,var/}run/runc/       rw,
+/{,var/}run/runc/**     mrwklix,
+
+# Wide read access to /proc, but somewhat limited writes for now
+@{PROC}/ r,
+@{PROC}/** r,
+@{PROC}/[0-9]*/attr/exec w,
+@{PROC}/sys/net/** w,
+@{PROC}/[0-9]*/cmdline r,
+@{PROC}/[0-9]*/oom_score_adj w,
+
+# Wide read access to /sys
+/sys/** r,
+# Limit cgroup writes a bit
+/sys/fs/cgroup/*/docker/   rw,
+/sys/fs/cgroup/*/docker/** rw,
+/sys/fs/cgroup/*/system.slice/   rw,
+/sys/fs/cgroup/*/system.slice/** rw,
+
+# Allow tracing other processes (especially ourselves)
+ptrace (read, trace),
```

Comment:
Aha:
```
Log: apparmor="DENIED" operation="ptrace" profile="snap.docker.dockerd" pid=18065 comm="docker-runc" requested_mask="trace" denied_mask="trace" peer="snap.docker.dockerd"
```

Adding just `ptrace (trace) peer=@{profile_name},` is sufficient, so that'll be in the next update. :+1: